linux_os/guide/system/accounts/accounts-pam/display_login_attempts/bash/shared.sh

```diff
@@ -1,52 +1,9 @@
-            CURRENT_PROFILE="custom/hardening"
-            # Ensure a backup before changing the profile
-            authselect apply-changes -b --backup=before-pwhistory-hardening.backup
-            authselect select $CURRENT_PROFILE
-            for feature in $ENABLED_FEATURES; do
-                authselect enable-feature $feature;
-            done
-        fi
-        # Include the desired configuration in the custom profile
-        CUSTOM_POSTLOGIN="/etc/authselect/$CURRENT_PROFILE/postlogin"
-        # The line should be included on the top of postlogin file
-        if [ "$(grep -c "^\s*session.*required.*pam_lastlog.so\s\+showfailed\s*$" $CUSTOM_POSTLOGIN)" -eq 0 ]; then
-            sed -i --follow-symlinks '0,/^session.*/s/^session.*/session     required                   pam_lastlog.so showfailed\n&/' $CUSTOM_POSTLOGIN
-        fi
-        if grep -q "^\s*session.*required.*pam_lastlog.so.*silent.*" $CUSTOM_POSTLOGIN; then
-            # remove 'silent' option
-            sed -i --follow-symlinks 's/^\(session.*required.*pam_lastlog.so\).*/\1 showfailed/g' $CUSTOM_POSTLOGIN
-        fi
-        authselect apply-changes -b --backup=after-pwhistory-hardening.backup
-    else
-        echo "
-authselect integrity check failed. Remediation aborted!
-This remediation could not be applied because the authselect profile is not intact.
-It is not recommended to manually edit the PAM files when authselect is available.
-In cases where the default authselect profile does not cover a specific demand, a custom authselect profile is recommended."
-        false
-    fi
-else
-    {{% if product in ["sle12", "sle15"] or 'ubuntu' in product %}}
-    {{% set pam_lastlog_path = "/etc/pam.d/login" %}}
-    {{% else %}}
-    {{% set pam_lastlog_path = "/etc/pam.d/postlogin" %}}
-    {{% endif %}}
-
-    if [ "$(grep -c "^\s*session.*required.*pam_lastlog.so\s\+showfailed\s*$" {{{ pam_lastlog_path }}})" -eq 0 ]; then
-        sed -i --follow-symlinks '0,/^session.*/s/^session.*/session     required                   pam_lastlog.so showfailed\n&/' {{{ pam_lastlog_path }}}
-    fi
-    if grep -q "^\s*session.*required.*pam_lastlog.so.*silent.*" {{{ pam_lastlog_path }}}; then
-        # remove 'silent' option
-        sed -i --follow-symlinks 's/^\(session.*required.*pam_lastlog.so\).*/\1 showfailed/g' {{{ pam_lastlog_path }}}
-    fi
-fi
+{{% if product in ["sle12", "sle15", "ubuntu"] %}}
+{{{ bash_ensure_pam_module_configuration('/etc/pam.d/login', 'session', 'required', 'pam_lastlog.so', 'showfailed', '', '^$') }}}
+{{{ bash_remove_pam_module_option_configuration('/etc/pam.d/login', 'session', 'required', 'pam_lastlog.so', 'silent') }}}
+{{% else %}}
+{{{ bash_ensure_pam_module_configuration('/etc/pam.d/postlogin', 'session', 'required', 'pam_lastlog.so', 'showfailed', '', '^$') }}}
+{{{ bash_remove_pam_module_option_configuration('/etc/pam.d/postlogin', 'session', 'required', 'pam_lastlog.so', 'silent') }}}
```

Comment:
During my researches I fairly remember a possible case where the line is present twice in the same group but with different control and I preferred to be more conservative here. However, I didn't find a real example now. I think the macro to remove an option make sense when the line where the option is supposed to be present is there. The `bash_ensure_pam_module_configuration` ensure that but in other cases, the `bash_ensure_pam_module_line` could be called directly, if necessary.

Maybe it would be better to accept a `regex` there instead of completely ignore the control.